### ISSUE TITLE
MaskedMimic problem

### ISSUE THREAD

**Issue by YunyangZhao** [LeCAR-Lab/ASAP#22]
Hello, I encountered an issue when running an SMPL file using the MaskedMimic method after obtaining the SMPL file through TRAM during the data preparation stage. The specific error message is:

(ProtoMotions) root@autodl-container-14944f8a6a-5354993b:~/autodl-tmp/ProtoMotions# python protomotions/train_agent.py +exp=full_body_tracker/transformer_flat_terrain +robot=smpl +simulator=isaacgym motion_file="/root/autodl-tmp/ProtoMotions/data/motions/hps_track_0.npy" +experiment_name=v1
Importing module 'gym_38' (/root/autodl-tmp/isaacgym/python/isaacgym/_bindings/linux-x86_64/gym_38.so)
Setting GYM_USD_PLUG_INFO_PATH to /root/autodl-tmp/isaacgym/python/isaacgym/_bindings/linux-x86_64/usd/plugInfo.json
2025-03-20 10:52:32,189 - INFO - logger - logger initialized
INFO:poselib:logger initialized
protomotions/train_agent.py:72: UserWarning: 
The version_base parameter is not specified.
Please specify a compatibility version level, or None.
Will assume defaults for version 1.1
  @hydra.main(config_path="config", config_name="base")
/root/miniconda3/envs/ProtoMotions/lib/python3.8/site-packages/hydra/_internal/defaults_list.py:251: UserWarning: In 'base': Defaults list is missing `_self_`. See https://hydra.cc/docs/1.2/upgrades/1.0_to_1.1/default_composition_order for more information
  warnings.warn(msg, UserWarning)
/root/miniconda3/envs/ProtoMotions/lib/python3.8/site-packages/hydra/_internal/hydra.py:119: UserWarning: Future Hydra versions will no longer change the working directory at job runtime by default.
See https://hydra.cc/docs/1.2/upgrades/1.1_to_1.2/changes_to_job_working_dir/ for more information.
  ret = run_job(
Initializing distributed: GLOBAL_RANK: 0, MEMBER: 1/1
[2025-03-20 10:52:33,006][lightning_fabric.utilities.rank_zero][INFO] - ----------------------------------------------------------------------------------------------------
distributed_backend=nccl
All distributed processes registered. Starting with 1 process
----------------------------------------------------------------------------------------------------

[rank: 0] Seed set to 0
Setting seed: 0
/root/miniconda3/envs/ProtoMotions/lib/python3.8/site-packages/torch/functional.py:507: UserWarning: torch.meshgrid: in an upcoming release, it will be required to pass the indexing argument. (Triggered internally at ../aten/src/ATen/native/TensorShape.cpp:3549.)
  return _VF.meshgrid(tensors, **kwargs)  # type: ignore[attr-defined]
Initial heightfield size: 2800x6450 = 18060000 vertices
Optimizing mesh by merging flat regions...
Optimized mesh vertex count: 98
Note: 'sim' is not a field in RobotConfig, it will be ignored
PyTorch version 2.2.0+cu121
Device count 1
/root/autodl-tmp/isaacgym/python/isaacgym/_bindings/src/gymtorch
Using /root/.cache/torch_extensions/py38_cu121 as PyTorch extensions root...
Emitting ninja build file /root/.cache/torch_extensions/py38_cu121/gymtorch/build.ninja...
Building extension module gymtorch...
Allowing ninja to set a default number of workers... (overridable by setting the environment variable MAX_JOBS=N)
ninja: no work to do.
Loading extension module gymtorch...
Not connected to PVD
+++ Using GPU PhysX
Physics Engine: PhysX
Physics Device: cuda:0
GPU Pipeline: enabled
Adding terrain
Terrain added
Creating 4096 environments... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:00
/root/autodl-tmp/ProtoMotions/protomotions/utils/motion_lib.py:128: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  torch.tensor(key_body_ids, dtype=torch.long, device=device),
Loading motions from yaml/npy file
Loading 1/1 motion files: /root/autodl-tmp/ProtoMotions/data/motions/hps_track_0.npy
Error executing job with overrides: ['+exp=full_body_tracker/transformer_flat_terrain', '+robot=smpl', '+simulator=isaacgym', 'motion_file=/root/autodl-tmp/ProtoMotions/data/motions/hps_track_0.npy', '+experiment_name=v1']
Error in call to target 'protomotions.envs.mimic.env.Mimic':
InstantiationException("Error in call to target 'protomotions.utils.motion_lib.MotionLib':\nKeyError('__name__')\nfull_key: config.motion_lib")
full_key: env

Set the environment variable HYDRA_FULL_ERROR=1 for a complete stack trace.
```

---

Have you encountered this issue before?

**Comment by YunyangZhao** [comment]
I used np.load to inspect the .npy file obtained from TRAM and the .npy file in ProtoMotions, and I noticed that their contents are different. I would like to ask the author if there is a data conversion process between these two steps.

**Comment by Lingling006688** [comment]
I’ve encountered the same issue. I’ve already processed my video file with TRAM and obtained a .npy file. I’d like to ask how I can convert it into a compatible .pkl file? I’ve tried many times, but I still can’t load the motion file. Thank you for your time in advance! @YunyangZhao
